src/rgw/rgw_rest_s3.cc

```diff
@@ -3525,26 +3557,47 @@ void RGWRestoreObj_ObjStore_S3::send_response()
 
   rgw::sal::Attrs attrs = s->object->get_attrs();
   auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+  auto expire_iter = attrs.find(RGW_ATTR_RESTORE_EXPIRY_DATE);
+  auto type_iter = attrs.find(RGW_ATTR_RESTORE_TYPE);
   rgw::sal::RGWRestoreStatus restore_status;
+  rgw::sal::RGWRestoreType rt;
+  ceph::real_time expiration_date ;
   if (attr_iter != attrs.end()) {
     bufferlist bl = attr_iter->second;
     auto iter = bl.cbegin();
     decode(restore_status, iter);
   }
-  ldpp_dout(this, 10) << "restore_status=" << restore_status << dendl;
+
+  if (expire_iter != attrs.end()) {
+    bufferlist bl = expire_iter->second;
+    auto iter = bl.cbegin();
+    decode(expiration_date, iter);
+  }
+
+  if (type_iter != attrs.end()) {
+    
+    bufferlist bl = type_iter->second;
+    auto iter = bl.cbegin();
+    decode(rt, iter);
+  }
   
-  if (attr_iter == attrs.end() || restore_status != rgw::sal::RGWRestoreStatus::None) {
+  if (attr_iter == attrs.end() || restore_status == rgw::sal::RGWRestoreStatus::RestoreFailed) {
     s->err.http_ret = 202; //Accepted
-    dump_header(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else if (restore_status != rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
+    dump_header(s, "x-amz-restore", "on-going-request=\"true\"");
+  } else if (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
     s->err.http_ret = 409; // Conflict
-    dump_header_if_nonempty(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else if (restore_status != rgw::sal::RGWRestoreStatus::CloudRestored) {
-    s->err.http_ret = 200; // OK
-    dump_header_if_nonempty(s, "x-amz-restore", rgw_bl_str(restore_status));
+    dump_header(s, "x-amz-restore", "on-going-request=\"true\"");
+  } else if (restore_status == rgw::sal::RGWRestoreStatus::CloudRestored) {
+    if (rt == rgw::sal::RGWRestoreType::Temporary) {
+    s->err.http_ret = 200; // OK temporary restore will have modified expiry date.
+    dump_header(s, "x-amz-restore", "ongoing-request=\"false\", expiry-date=\""+ dump_time_to_str(expiration_date) +"\"");
```

Comment:
expiry_date attr is stored in string format. We need to convert it to real_time before calling dump_time_to_str().. refer to "parse_rgwx_mtime".

Reply:
Will check.